personalrobotics/ada_feeding
# Move2Plan/Execute (MoveTo Behavior Overhaul)

# Description

Adds the following functionality / overhauled behaviors:

* MoveIt2Plan: given a bunch of planning parameters returns a JointTrajectory
* * If all goal constraints are satisfied, return None and Success
* * If any path constraint is unsatisfied, return None and Failure (unless explicitly ignored)
* MoveIt2XConstraint: compile blackboard objects into the constraint kwargs dictionary expected by MoveIt2Plan
* MoveIt2Execute: Given a JointTrajectory, execute it on the robot
* Add the option for setting the debug log_level at the launch file level
* Add the default action_select policy to the ada_feeding main launch file
* Extend the AcquireFood tree to carry out MoveAbove in order to test the basic functionality of the new behaviors.

# Testing procedure

Follow the testing procedure in #102 with the following additional steps:
* Verify that the robot moves 5cm above the shown `food` frame with the tines perpendicular to the X (red) axis (i.e. constant action 0)
* Alternate between the 2 carrot images and see the robot go back and forth moving above the two foods

_Note: will be happy to take recommendations on how to test other components of the MoveIt2 behaviors, but this is sufficient for me now to continue working on the AcquireFood tree_

# Before opening a pull request

# Before Merging
